setup.py

```diff
@@ -52,6 +52,13 @@ def __getitem__(self, item):
 def get_sphinx_command():
     """Lazy load of Sphinx distutils command class
     """
+    # If nbconvert is installed to .eggs on the fly when running setup.py then
+    # templates from it will not be found. This is a workaround.
+    if 'JUPYTER_PATH' not in os.environ:
+        import nbconvert
+        os.environ['JUPYTER_PATH'] = os.path.realpath(os.path.join(os.path.dirname(nbconvert.__file__), '..', 'share', 'jupyter'))
+        print("Setting JUPYTER_PATH={}".format(os.environ['JUPYTER_PATH']))
```

Comment:
`note`: Are these "temporary" workarounds for newer version or permanent fixes? if it's temporary and issue exist somewhere, link would be nice.

Reply:
My impression is that they changed the logic here around 6.0.0 and that it is a bit fragile, particularly when it is not used as a "system" install (e.g. Spack, with packages in their own prefixes, and with the Python stuff that also installs dependencies on the fly to separate prefixes under `.eggs/`). I came across https://github.com/spack/spack/pull/21827, for example. I did not find any particularly good reference though...

Reply:
ok, thanks for clarification.